exts/betterTooltip/betterTooltip.js

```diff
@@ -0,0 +1,537 @@
+                var series      = dimensions.series; 
+                var seriesValue = scene.getSeries();
+
+                var rootColor   = scene.root.panel().axes.color;
+                var colorVar    = scene.vars.color;
+                var color       = rootColor.isDiscrete() && colorVar ? rootColor.scale(colorVar).color : null;
+
+                var value;
+                if(seriesValue != null) {
+                    value = {
+                        value: seriesValue,
+                        label: scene.getSeriesLabel()
+                    };
+                } else {
+                    var firstMeasure = tooltipModels.measures[0]
+                    value = {value: firstMeasure.value, label: ""};
+
+                    if(tooltipModels.category && (!series || tooltipModels.measures.length > 1)) {
+                        value.label = tooltipModels.category.value.label;
+                        tooltipModels.category = null;
+                    } else value.label = firstMeasure.label;
+                }
+
+                return {color: color, value: value, label: (series && series.label) || ""};
+            }
+        }
+
+        /**
+         * The default format function replaces tokens in the received format string
+         * with values from the current tooltipModel subject (category, series, measure).
+         *
+         * @param {Object} tooltipModel The tooltip model object (not used by the default formater).
+         * @param {Object} subject The current tooltip subject being formated.
+         * @param {string} formatString The format string.
+         *
+         * @return {string} The formated text.
+         */
+        function defaultFormatFunction(tooltipModel, subject, formatString) {
+            var matcher = /\{(.*?)\}/g;
+
+            var result = formatString;
+
+            var match;
+            while(match = matcher.exec(formatString)) {
+                var alternatives = match[1].split("|");
+
+                while(alternatives.length) {
+                    var path = alternatives.shift().split(".");
+
+                    var value = subject;
+                    while(value && path.length) {
+                        value = value[path.shift()];
+                    }
+
+                    if(value) {
+                        result = result.replace(match[0], def.html.escape(value.toString()));
+                        // skip remaining alternatives
+                        break;
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        /**
+         * The renderer method is called with a pre-built tooltip model object,
+         * and having the formater instance as `this` context.
+         *
+         * @param {Object} tooltipModel The tooltip model object.
+         *
+         * @return {string} The HTML of the rendered tooltip.
+         */
+        function betterTooltipRenderer(tooltipModel) {
```

Comment:
From my understand the series and measures render could be refactored out for readability.